Grammar/python.gram

```diff
@@ -1221,20 +1230,32 @@ invalid_if_stmt:
     | 'if' named_expression NEWLINE { RAISE_SYNTAX_ERROR("expected ':'") }
     | a='if' a=named_expression ':' NEWLINE !INDENT {
         RAISE_INDENTATION_ERROR("expected an indented block after 'if' statement on line %d", a->lineno) }
+    | 'if' named_expression ':' (NEWLINE INDENT statements | simple_stmts) a='elif' {
+        RAISE_SYNTAX_ERROR_STARTING_FROM(a, "'elif' does not match a 'if' here. Maybe try unindenting?") }
+    | 'if' named_expression ':' (NEWLINE INDENT statements | simple_stmts) a='else' {
+        RAISE_SYNTAX_ERROR_STARTING_FROM(a, "'else' does not match a 'if'/'while'/'for'/'except' here. Maybe try unindenting?") }
 invalid_elif_stmt:
     | 'elif' named_expression NEWLINE { RAISE_SYNTAX_ERROR("expected ':'") }
     | a='elif' named_expression ':' NEWLINE !INDENT {
         RAISE_INDENTATION_ERROR("expected an indented block after 'elif' statement on line %d", a->lineno) }
+    | 'elif' named_expression ':' (NEWLINE INDENT statements | simple_stmts) a='elif' {
+        RAISE_SYNTAX_ERROR_STARTING_FROM(a, "'elif' does not match a 'if' here. Maybe try unindenting?") }
+    | 'elif' named_expression ':' (NEWLINE INDENT statements | simple_stmts) a='else' {
```

Comment:
I an not particularly convinced that the message is generic regarding the match ('if'/'while'/'for'/'except')